DIPs/DIPxxxx.md

```diff
@@ -0,0 +1,233 @@
+
+struct SomeStruct {
+    void someFunction() {
+        // this function inherits `@safe` but doesn't inherit
+        // `nothrow`, `pure`, nor `@nogc`
+
+        throw new Exception(); // not `nothrow` or `@nogc`
+        __gshared int a;       // not `pure`
+    }
+}
+
+void someTemplateFunction(T...)(T args) {
+    // loses auto inference
+    // this function now is *required* to be `@safe nothrow pure @nogc`
+    // even though it could auto infer these attributes
+    // this usually isn't the desired behavior to have for a template
+}
+
+struct SomeBestPracticeStruct {
+// these are *required* in each and every new aggregate scope
+// for these attributes to be continued as the "default"
+nothrow pure @nogc:
+
+    void someFunction() {
+        // now we have the desired effect, SomeBestPracticeStruct.someFunction
+        // is `@safe nothrow pure @nogc` in comparison to SomeStruct.someFunction
+        // that is only `@safe`
+    }
+}
+
+struct SomeOtherBEsstPractiseStruct {
+pure:
+
+    void oneOrTwofunctionThatThrows() {
+        // now we need just one or two functions that might need to throw
+        // there are no inverses to `nothrow` nor `@nogc` so we would
+        // need to shuffle this to the top
+
+        throw new Exception();
+    }
+
+// now the intention isn't as obvious as this is separate from `pure:`
+// and does not appear at the start of the aggregate scope
+nothrow @nogc:
+
+    // ...
+
+}
+
+```
+
+To outline the problems with attempting to apply different default attributes:
+* Not all attributes flow through aggregates, only `@safe`, `@trusted`, and `@system` do so. This means `@attribute:` is a no solution for every attribute. Changing this behavior would also cause significant breaking changes.
+* Templates lose auto inference. As the attributes are applied to *everything* in the scope. This means they are also applied to templates and those templates are thus required to follow the rules of those attributes. Since templates can auto infer these attributes, the desired effect would be to simply have the template continue to be auto inferred. The function calling the template would then enforce any rules that would need to be applied.
+* Attribute bloat at the aggregate scope. Since some attributes don't propagate through all scopes they must be reapplied for each new aggregate scope. This can become tedious and add extra bloat where it shouldn't be necessary.
+* Lack of inverse effects for function attributes leads to odd reordering instead of maintaining the logical order of function definitions affecting readability. The `@attribute:` then appears somewhere in the middle of a large amount of code making it less visible and apparent of what the intention is.
+
+## Prior Work
+
+[DIP1029: Add throw as a function attribute.](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1029.md)
+
+[DIP10XX: Make nothrow the default.](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1029.md)
```

Comment:
The link points to DIP1029 as well. I assume this is in anticipation of Walter's DIP?